controllers/v1/programs.js

```diff
@@ -0,0 +1,113 @@
+   * @apiParamExample {json} Request-Body:
+* {
+  "entityId": "",
+  "entityType": ""
+* }
+   * @apiSuccessExample {json} Success-Response:
+*     HTTP/1.1 200 OK
+*     {
+       "result": true,
+       "data": [
+           {
+               "programId": "",
+               "programName": "",
+               "type": ""
+           }
+       ]
+*     }
+   * @apiUse errorBody
+   */
+
+exports.list = async function (req, res) {
+  
+    if (!req.body.entityId || !req.body.entityType) {
+        res.status(400);
+        var response = {
+            result: false,
+            message: 'entityId and entityType are required fields'
+        }
+        res.send(response);
+    }
+    else {
+         
+         //get quey from cassandra
+         model.MyModel.findOneAsync({ qid: "programs_list_query" }, { allow_filtering: true })
+         .then(async function (result) {
+
+             let bodyParam = JSON.parse(result.query);
+
+             let id = await getCreatedByField(req,res);
+
+             bodyParam.filter.fields[0].dimension = req.body.entityType;
+             bodyParam.filter.fields[0].value = req.body.entityId;
+
+             //fetch assessment programs from assessment datasource
+             if (config.druid.assessment_datasource_name) {
+                 bodyParam.dataSource = config.druid.assessment_datasource_name;
+             }
+             
+             bodyParam.filter.fields[1].fields[0].fields[0].dimension = "userId";
+             bodyParam.filter.fields[1].fields[0].fields[0].value = id;
+
+             let options = config.druid.options;
+             options.method = "POST";
+             options.body = bodyParam;
+
+             let assessmentPrograms = await rp(options);
+
+             //fetch observation programs from observation datasource
+             if (config.druid.observation_datasource_name) {
+                 bodyParam.dataSource = config.druid.observation_datasource_name;
+             }
+
+             bodyParam.filter.fields[1].fields[0].fields[0].dimension = "createdBy";
+             options.body = bodyParam;
+             let observationPrograms = await rp(options);
+             
+             let programs = [...assessmentPrograms,...observationPrograms];
+
+             if (!programs.length) {
+                 res.send({ "result": false, "data": [] });
+             }
+             else {
+                 let response = await helperFunc.programsListCreation(programs);
+                 res.send(response);
+             }
+         })
+            .catch(function (err) {
+                console.log(err);
+                res.status(500);
+                var response = {
+                    result: false,
+                    message: 'INTERNAL_SERVER_ERROR'
+                }
+                res.send(response);
+            })
+        }
+};
+
+
+// Function for getting createdBy field from header access token
+async function getCreatedByField(req, res) {
```

Comment:
@deepakr123 - No need to validate with keycloak, set it in the request body after request is autheticated in the middleware.

Reply:
resolved